internal/k8s/status.go

```diff
@@ -109,6 +111,9 @@ func (suh *StatusUpdateHandler) Start(ctx context.Context) error {
 	suh.log.Info("started status update handler")
 	defer suh.log.Info("stopped status update handler")
 
+	// Enable StatusUpdaters to start sending updates to this handler.
+	close(suh.sendUpdates)
```

Comment:
What would happen if this instance looses the leader and a new one is elected. Would't you have 2 updates to the status now?

Reply:
if this instances loses leadership the process will shut down so only one should be running, the context governing the runnable should be cancelled